From 70d5f355f317def7fadc993ec97fd2e286095486 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Tue, 28 May 2024 14:46:40 +0530
Subject: [PATCH] ath12k: remove host2rxdma interrupt handling for
 QCN9274/IPQ5332/QCN6432

QCN9274/IPQ5332/QCN6432 firmware lacks support for backpressure
interrupts. Due to shared ring mask with the WBM error path and REO
exception path, the interrupt handler is triggered needlessly. To
prevent unnecessary ring replenishment, removing this interrupt mask for
above mentioned chipsets.

Patch-dependency : 710-004-ath12k-Add-and-initialize-hw-params-for-IPQ5332.patch,
		   822-ath12k-qcn6432-bring-up.patch

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hw.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/hw.c b/drivers/net/wireless/ath/ath12k/hw.c
index 6da95ad..16656ef 100644
--- a/drivers/net/wireless/ath/ath12k/hw.c
+++ b/drivers/net/wireless/ath/ath12k/hw.c
@@ -1047,12 +1047,6 @@ static struct ath12k_hw_ring_mask ath12k_hw_ring_mask_qcn9274 = {
 		0, 0, 0, 0, 0, 0, 0, 0,
 		0, 0, 0
 	},
-	.host2rxdma = {
-		0, 0, 0,
-		ATH12K_HOST2RXDMA_RING_MASK_0,
-		0, 0, 0, 0, 0, 0, 0, 0,
-		0, 0, 0
-	},
 	.tx_mon_dest = {
 		ATH12K_TX_MON_RING_MASK_0,
 		ATH12K_TX_MON_RING_MASK_1,
@@ -1115,11 +1109,6 @@ static struct ath12k_hw_ring_mask ath12k_hw_ring_mask_ipq5332 = {
 		ATH12K_REO_STATUS_RING_MASK_0,
 		0, 0, 0, 0, 0, 0, 0, 0,
 	},
-	.host2rxdma = {
-		0, 0, 0,
-		ATH12K_HOST2RXDMA_RING_MASK_0,
-		0, 0, 0, 0, 0, 0, 0, 0,
-	},
 	.tx_mon_dest = {
 		ATH12K_TX_MON_RING_MASK_0,
 		ATH12K_TX_MON_RING_MASK_1,
@@ -1193,13 +1182,6 @@ static struct ath12k_hw_ring_mask ath12k_hw_ring_mask_qcn6432 = {
 		ATH12K_REO_STATUS_RING_MASK_0,
 		0, 0, 0, 0, 0, 0, 0, 0, 0,
 	},
-	.host2rxdma = {
-		0, 0,
-		ATH12K_HOST2RXDMA_RING_MASK_0,
-		ATH12K_HOST2RXDMA_RING_MASK_1,
-		ATH12K_HOST2RXDMA_RING_MASK_2,
-		0, 0, 0, 0, 0, 0, 0,
-	},
 	.tx_mon_dest = {
 		ATH12K_TX_MON_RING_MASK_0,
 		ATH12K_TX_MON_RING_MASK_1,
-- 
2.34.1

